File: backend-api/src/main/java/com/smartenergy/backendapi/model/Usuario.java
package com.smartenergy.backendapi.model;

import jakarta.persistence.Entity;
import jakarta.persistence.Inheritance;
import jakarta.persistence.InheritanceType;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Inheritance(strategy = InheritanceType.JOINED)
@Table(name = "Usuario")
public abstract class Usuario extends BaseEntity {
    private boolean activo;
    private String nombre;
    private String apellido;
    private String cif;
    private String password;

}

File: mobile-app/app/src/main/java/com/example/smartenergy/model/Usuario.kt
package com.example.smartenergy.model

enum class Rol {
    ADMINISTRADOR, APOYO_LOGISTICO
}

interface Usuario {
    val id: String?
    val nombre: String
    val apellido: String
    val cif: String
    val password: String
    val rol: Rol
    val activo: Boolean
}

data class ApoyoLogistico(
    override val id: String? = null,
    override val nombre: String,
    override val apellido: String,
    override val cif: String,
    override val password: String,
    override val activo: Boolean
) : Usuario {
    override val rol: Rol get() = Rol.APOYO_LOGISTICO
}

data class Administrador(
    override val id: String? = null,
    override val nombre: String,
    override val apellido: String,
    override val cif: String,
    override val password: String,
    override val activo: Boolean,
    val nivelAcceso: String
) : Usuario {
    override val rol: Rol get() = Rol.ADMINISTRADOR
}
